dcos/package.py

```diff
@@ -85,11 +85,17 @@ def uninstall(pkg, package_name, remove_all, app_id, cli, app):
         cli = app = True
 
     uninstalled = False
-    if cli:
+    installed = installed_packages(pkg, app_id, package_name)
+    installed_cli = next((True for installed_pkg in installed
+                          if installed_pkg.get("command")), False)
+    installed_app = next((True for installed_pkg in installed
+                          if installed_pkg.get("apps")), False)
```

Comment:
cosmos returns an error if a package was not installed, but it doesn't know anything about the cli. We don't want to error when uninstalling if only the cli is installed. So only try to uninstall the components that are returned from list.